dojo-dungeon/src/dojo/compiler/DojoCompiler.java

```diff
@@ -320,24 +253,93 @@ private boolean stage_6_checkSecondOutput() {
     return true;
   }
 
-  private String getSource(String fn) throws IOException {
-    return Files.readString(Paths.get(fn));
-  }
+  private Class<?> compile(String pathToSourceFiles, String fqClassName) throws Exception {
+    Class<?> cls = Class.forName(fqClassName);
 
-  private Class<?> compile(String source, String className) throws Exception {
-    // Save source text to temporary file
-    File root = Files.createTempDirectory("java").toFile();
-    File sourceFile = new File(root, className + ".java");
-    assert sourceFile.getParentFile().mkdirs();
-    Files.writeString(sourceFile.toPath(), source);
+    // hard coded build dir for now:
+    String argBuildDir = "build/temp";
+
+    String argToCompile = Paths.get(pathToSourceFiles, cls.getSimpleName() + ".java").toString();
+    URL argToLoad = Paths.get(argBuildDir).toUri().toURL();
+    Logger logger = Logger.getLogger(DojoCompiler.class.getName());
+    logger.warning(
+        "Compiling: "
+            + fqClassName
+            + " in: "
+            + argToCompile
+            + " ("
+            + argBuildDir
+            + ") and load: "
+            + argToLoad);
 
     // Compile source file
     JavaCompiler compiler = ToolProvider.getSystemJavaCompiler();
-    compiler.run(null, null, null, sourceFile.getPath());
+    compiler.run(null, null, null, "-d", argBuildDir, argToCompile);
 
     // Load compiled class
-    URLClassLoader classLoader = URLClassLoader.newInstance(new URL[] {root.toURI().toURL()});
+    try (URLClassLoader reloadClassLoader =
+        new URLClassLoader(new URL[] {argToLoad}, cls.getClassLoader().getParent())) {
+      return reloadClassLoader.loadClass(fqClassName);
+    }
+  }
+
+  /**
+   * Compile and load an existing class that depends on other classes (imports them).
+   *
+   * @param pathToSourceFiles the path to the source file.
+   * @param fqClassName the fully qualified name of the class.
+   * @return the loaded class.
+   * @throws Exception if compiling or loading fails.
+   */
+  public static Class<?> compileClassDependentOnOthers(String pathToSourceFiles, String fqClassName)
+      throws Exception {
+    Class<?> cls = Class.forName(fqClassName);
+
+    // hard coded build dir for now:
+    String argBuildDir = "build/temp";
```

Comment:
@cagix Hättest du hierfür eine Idee? Zurzeit ist das Verzeichnis "fest" - und "build/tmp" gibt es bereits, deshalb "build/temp".

Reply:
wo unterhalb von "build/" wird das file denn ursprünglich kompiliert? das wäre der natürliche platz...

Reply:
Na ja, der `DojoStarter` zum Beispiel unter `dojo-dungeon/build/classes/java/main/starter/DojoStarter.class`, also wäre es `build/classes/java/main`.

Die Ordnerstruktur wird durch die `-d`-Option automatisch erzeugt.

Bin aber noch nicht sicher, ob das Überschreiben einfach so möglich wäre, oder ob da Java ein "handle" oder ähnliches drauf hält ...

---

Zudem muss ich den Packages noch bessere Namen geben.

Reply:
Handle könnte gut sein - allerdings ist zum Zeitpunkt der Ausführung alles vorbei: es wird nicht mehr geschrieben (kompiliert), und an sich auch nicht mehr gelesen... 

(seltsam: wieso zeigt mir die "blaue glocke" an, dass hier aktualisierungen sind, aber unter "news" taucht der pr nicht mit auf? jetzt schon mind. 3x ... hmmm.)

Reply:
> (seltsam: wieso zeigt mir die "blaue glocke" an, dass hier aktualisierungen sind, aber unter "news" taucht der pr nicht mit auf? jetzt schon mind. 3x ... hmmm.)

Bei mir das Gleiche xD "blaue Glocke" an und "Inbox 1", aber kein Eintrag hierher. ;) Möglicherweise ein Bug.

---

Oki, dann könnte das so funktionieren. :) Ich ändere das morgen früh.

Reply:
Es hat geklappt 👍🏼

Reply:
@tgrothe das klingt gut. damit ich den überblick nicht verliere, habe ich in meinem review-kommentar (https://github.com/Dungeon-CampusMinden/Dungeon/pull/1532#pullrequestreview-2091168395) checkboxen aus den stichpunkten gemacht. kannst du bitte dort abhaken, was schon umgesetzt ist?